src/core/io/writer/h5md_core.cpp

```diff
@@ -48,56 +49,42 @@ static void backup_file(const std::string &from, const std::string &to) {
 }
 
 static std::vector<hsize_t> create_dims(hsize_t dim, hsize_t size) {
-  if (dim == 3)
+  switch (dim) {
+  case 3:
     return std::vector<hsize_t>{size, size, dim};
-  if (dim == 2)
+  case 2:
     return std::vector<hsize_t>{size, size};
-  if (dim == 1)
+  case 1:
     return std::vector<hsize_t>{size};
-
-  throw std::runtime_error(
-      "H5MD Error: datasets with this dimension are not implemented\n");
+  default:
+    throw std::runtime_error(
+        "H5MD Error: datasets with this dimension are not implemented\n");
+  }
 }
 
-// Correct Chunking is important for the IO performance!
-std::vector<hsize_t> File::create_chunk_dims(hsize_t dim, hsize_t size,
-                                             hsize_t chunk_size) {
-  if (dim == 3)
-    return std::vector<hsize_t>{chunk_size, size, dim};
-  if (dim == 2)
-    return std::vector<hsize_t>{chunk_size, size};
-  if (dim == 1)
+// Correct chunking is important for the IO performance!
+std::vector<hsize_t> File::create_chunk_dims(hsize_t dim, hsize_t size) {
+  switch (dim) {
+  case 3:
+    return std::vector<hsize_t>{1, size, dim};
+  case 2:
+    return std::vector<hsize_t>{1, size};
+  case 1:
     return std::vector<hsize_t>{size};
-
-  throw std::runtime_error(
-      "H5MD Error: datasets with this dimension are not implemented\n");
+  default:
+    throw std::runtime_error(
+        "H5MD Error: datasets with this dimension are not implemented\n");
+  }
 }
+
 static std::vector<hsize_t> create_maxdims(hsize_t dim) {
-  if (dim == 3)
-    return std::vector<hsize_t>{H5S_UNLIMITED, H5S_UNLIMITED, H5S_UNLIMITED};
-  if (dim == 2)
-    return std::vector<hsize_t>{H5S_UNLIMITED, H5S_UNLIMITED};
-  if (dim == 1)
-    return std::vector<hsize_t>{H5S_UNLIMITED};
-
-  throw std::runtime_error(
-      "H5MD Error: datasets with this dimension are not implemented\n");
+  return std::vector<hsize_t>(dim, H5S_UNLIMITED);
 }
 
 /* Initialize the file related variables after parameters have been set. */
-void File::InitFile() {
+void File::init_file() {
   m_backup_filename = m_filename + ".bak";
-  // use a separate mpi communicator if we want to write out ordered data. This
-  // is in order to avoid  blocking by collective functions
-  auto world = boost::mpi::communicator();
-
-  if (m_write_ordered)
-    m_hdf5_comm = world.split(world.rank(), 0);
-  else
-    m_hdf5_comm = world;
-
-  if (m_write_ordered && this_node != 0)
-    return;
+  m_hdf5_comm = boost::mpi::communicator();
```

Comment:
Since this is fixed now you can initialize it in the constructor, should also probably be a passed in, so that the client can chose which nodes participate...